library/core/src/ffi/c_str.rs

```diff
@@ -504,6 +507,13 @@ impl CStr {
         self.inner.as_ptr()
     }
 
+    /// We could eventually expose this publicly, if we wanted.
+    #[inline]
+    #[must_use]
+    const fn as_non_null_ptr(&self) -> NonNull<c_char> {
+        NonNull::from(&self.inner).as_non_null_ptr()
+    }
+
```

Comment:
Decided to put this because it seems like something we'd probably add in the future, but I don't really feel like doing an ACP for this right now. It felt reasonable enough to factor into its own method, at least.

Reply:
I'm not sure what you're referring to -- `CStr::as_ptr()` just calls the slice `as_ptr()`, which is just a couple pointer casts. I only found `addr_of!` used in `Rc`/`Arc::as_ptr`, and those are digging into a further heap allocation.

Reply:
Oh, sorry, I totally mixed up the implementations for `as_ptr` and `to_bytes_with_nul`. The latter uses `addr_of` to convert the slice into `*const [u8]` instead of the existing method, but that's going to be unsafe anyway, so, it's fair.

Reply:
Huh, I don't see any reason it's *needed* there either, just a little bit shorter than it was with double pointer casts before `addr_of!`. No matter.